packages/optimizer/lib/transformers/ReorderHeadTransformer.js

```diff
@@ -41,33 +41,30 @@ class HeadNodes {
   }
 
   uniquifyAndSortCustomElements() {
-    this._scriptRenderDelayingExtensions = this._removeDuplicateCustomExtensions(
+    this._scriptRenderDelayingExtensions = this._sortExtensions(
       this._scriptRenderDelayingExtensions
     );
-    this._scriptNonRenderDelayingExtensions = this._removeDuplicateCustomExtensions(
+    this._scriptNonRenderDelayingExtensions = this._sortExtensions(
       this._scriptNonRenderDelayingExtensions
     );
   }
 
-  _removeDuplicateCustomExtensions(extensions) {
-    const nodesByName = new Map();
-    for (const node of extensions) {
-      const name = this._getName(node);
-      nodesByName.set(name, node);
-    }
-    return Array.from(nodesByName.values());
+  _sortExtensions(extensions) {
+    const sortedExtensions = new Map([...extensions].sort((a, b) => a[0].localeCompare(b[0])));
+    // TODO replace with Array#flat once Node 10 is EOL
+    return [].concat.apply([], Array.from(sortedExtensions.values()));
   }
 
   appendToHead(head) {
     appendChild(head, this._metaCharset);
+    appendAll(head, this._metaOther);
+    appendAll(head, this._resourceHintLinks);
```

Comment:
The `metaOther` can potentially includes lots of non-critical stuff like OpenGraph tags. I think in terms of performance, the resource hints should come right after the charset and before these other meta tags.

Reply:
good point! done